spec/scry/symbol_spec.cr

```diff
@@ -2,81 +2,75 @@ require "../spec_helper"
-    it "returns a ResponseMessage" do
-      text_document = TextDocument.new("inmemory://model/3", [""])
-      processor = SymbolProcessor.new(text_document)
-      processor.run.is_a?(ResponseMessage).should be_true
-    end
-
     it "contains SymbolInformation" do
       text_document = TextDocument.new("uri", [""])
       processor = SymbolProcessor.new(text_document)
       response = processor.run
-      response.result.is_a?(Array(SymbolInformation)).should be_true
+      response.is_a?(Array(SymbolInformation)).should be_true
     end
 
     it "returns Class symbols" do
       text_document = TextDocument.new("uri", ["class Test; end"])
       processor = SymbolProcessor.new(text_document)
-      response = processor.run
-      result = response.result.as(Array(SymbolInformation)).first
+      symbols = processor.run
+      result = symbols.first
       result.kind.is_a?(SymbolKind::Class).should be_true
     end
 
     it "returns Struct symbols as a Class" do
       text_document = TextDocument.new("uri", ["struct Test; end"])
       processor = SymbolProcessor.new(text_document)
-      response = processor.run
-      result = response.result.as(Array(SymbolInformation)).first
+      symbols = processor.run
+      result = symbols.first
       result.kind.is_a?(SymbolKind::Class).should be_true
     end
 
     it "returns Module symbols" do
       text_document = TextDocument.new("uri", ["module Test; end"])
       processor = SymbolProcessor.new(text_document)
-      response = processor.run
-      result = response.result.as(Array(SymbolInformation)).first
+      symbols = processor.run
+      result = symbols.first
       result.kind.is_a?(SymbolKind::Module).should be_true
     end
 
     it "returns Method symbols" do
       text_document = TextDocument.new("uri", ["def test; end"])
       processor = SymbolProcessor.new(text_document)
-      response = processor.run
-      result = response.result.as(Array(SymbolInformation)).first
+      symbols = processor.run
+      result = symbols.first
       result.kind.is_a?(SymbolKind::Method).should be_true
     end
 
     it "returns instance vars as Variable symbols" do
       text_document = TextDocument.new("uri", ["@bar = nil"])
       processor = SymbolProcessor.new(text_document)
-      response = processor.run
-      result = response.result.as(Array(SymbolInformation)).first
+      symbols = processor.run
+      result = symbols.first
       result.as(SymbolInformation).kind.is_a?(SymbolKind::Variable).should be_true
     end
 
     describe "Property" do
       it "returns getters as Property symbols" do
         text_document = TextDocument.new("uri", ["class Foo; getter bar : Nil; end"])
         processor = SymbolProcessor.new(text_document)
-        response = processor.run
-        result = response.result.as(Array(SymbolInformation)).try { |r| r[1] }
+        symbols = processor.run
+        result = symbols.as(Array(SymbolInformation)).try { |r| r[1] }
```

Comment:
Are you sure you need the cast? (ditto below)

Reply:
Yeah this is old code, I can probably clean this up as well now

Reply:
Yes like below, using `symbols.first` should be enough